test/Altinn.App.Core.Tests/Infrastructure/Clients/Profile/ProfileClientTests.cs

```diff
@@ -0,0 +1,190 @@
+            };
+        appMetadataMock.Setup(am => am.GetApplicationMetadata()).ReturnsAsync(appMetadata);
+        services.AddSingleton(appMetadataMock.Object);
+
+        var tokenGenerator = new Mock<IAccessTokenGenerator>();
+        tokenGenerator.Setup(t => t.GenerateAccessToken("ttd", "test")).Returns("access-token");
+        services.AddSingleton(tokenGenerator.Object);
+
+        services.AddTelemetrySink();
+        services.AddSingleton<ILogger<ProfileClient>>(NullLogger<ProfileClient>.Instance);
+        services.Configure<PlatformSettings>(_ => { });
+        services.Configure<AppSettings>(o => o.RuntimeCookieName = "AltinnStudioRuntime");
+        services.Configure<CacheSettings>(o => o.ProfileCacheLifetimeSeconds = 1);
+
+        var handler = new Mock<HttpMessageHandler>();
+        handler
+            .Protected()
+            .Setup<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.IsAny<HttpRequestMessage>(),
+                ItExpr.IsAny<CancellationToken>()
+            )
+            .ReturnsAsync(() =>
+            {
+                var profile = userProfileFactory();
+                return new HttpResponseMessage
+                {
+                    StatusCode = profile is null ? HttpStatusCode.NotFound : HttpStatusCode.OK,
+                    Content = new StringContent(JsonSerializer.Serialize(profile))
+                };
+            })
+            .Verifiable();
+        var httpClient = new HttpClient(handler.Object);
+        services.AddSingleton(httpClient);
+        services.AddSingleton<IProfileClient, ProfileClient>();
+        return new(services.BuildServiceProvider(), handler);
+    }
+
+    [Fact]
+    public async Task Builds_From_DI_Container()
+    {
+        await using var fixture = BuildFixture();
+
+        var client = fixture.ServiceProvider.GetRequiredService<IProfileClient>();
+        Assert.NotNull(client);
+    }
+
+    [Fact]
+    public async Task Returns_Test_Profile()
+    {
+        const int userId = 1234;
+        TelemetrySink telemetry;
+        {
+            await using var fixture = BuildFixture(() => new UserProfile { UserId = userId, });
+            telemetry = fixture.ServiceProvider.GetRequiredService<TelemetrySink>();
+
+            var client = fixture.ServiceProvider.GetRequiredService<IProfileClient>();
+
+            var profile = await client.GetUserProfile(userId);
+            Assert.NotNull(profile);
+            Assert.Equal(userId, profile.UserId);
+        }
+
+        await Verify(telemetry.GetSnapshot());
+    }
+
+    [Fact]
+    public async Task Returns_Test_Profile_Cached()
+    {
+        const int userId = 1234;
+        await using var fixture = BuildFixture(() => new UserProfile { UserId = userId, });
+
+        var client = fixture.ServiceProvider.GetRequiredService<IProfileClient>();
+
+        var profile1 = await client.GetUserProfile(userId);
+        var profile2 = await client.GetUserProfile(userId);
```

Comment:
Resolve the client from different scopes to ensure that it caches across scopes.